dataService.js

```diff
@@ -0,0 +1,78 @@
+// Maps
+let productMap = {};
+let categoryMap = {};
+let combinedProductMap = null;
+
+// Method to create map of products
+function getProducts() {
+
+  const productData = require('./src/data/products.json');
```

Comment:
Load it when `productMap` is empty. No need to load it always. 
And loop and create it when `productMap` is empty also 

Apply this to all other methods

Reply:
Finally understood it, thanks it was a great suggestation, didn't know I can do that this way :)